subsys/mpsl/cx/Kconfig

```diff
@@ -15,6 +15,10 @@ config MPSL_CX
 
 if MPSL_CX
 
+config MPSL_CX_BT_INIT_ON_BOOT
```

Comment:
Maybe it would be a good idea to have a single `MPSL_CX_INIT_ON_BOOT` for both BT and zigbee/thread coex ? What do you think @jciupis ?

Reply:
To be honest, I don't see the point of this configuration option. I'm not aware of any use cases where it would be useful for the `mpsl_cx_thread` and `mpsl_cx_generic_3pin` plugins. Could we synchronize on why it is needed?

Reply:
In the wifi coex test we initialise the coex from the test side. 
Calling mpsl_cx_init() cause an issue, PPI channel are initialised twice. 
This is why I implement the option to not init mpsl

Reply:
It can also be used to enable or disable coex codepaths in the stack at runtime, if the application needs to for some reason.

Reply:
> It can also be used to enable or disable coex codepaths in the stack at runtime, if the application needs to for some reason.

As far as I understand, Coex is a feature that either must be present, if the device has an external Coexistence interface connected, or it is not needed at all. Enabling and disabling its support in runtime sounds like a very marginal use case, because the need for this feature depends on the hardware, which is fixed in runtime. Please correct me if I'm missing some scenarios here.

> In the wifi coex test we initialise the coex from the test side.

I'd prefer to avoid adding a Kconfig option that is meant only for internal testing and breaks the solution if modified by the user. Would it be possible to fix the initialization procedure inside the test?

Reply:
The option is enabled by default, it will be transparent for the user. It's tricky for us to change the behavior of the test.

Reply:
OK, I understand. I'd only suggest adding a dependency to `MPSL_CX_BT_INIT_ON_BOOT` then:
```
depends on MPSL_CX_BT_1WIRE || MPSL_CX_BT_3WIRE
```

Reply:
Discussed offline. We want to remove ` MPSL_CX_BT_INIT_ON_BOOT` as there is no customer use case for it

Reply:
Outdated